src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -804,8 +804,43 @@ protected bool TakeStartLine(SocketInput input)
                 QueryString = queryString;
                 HttpVersion = httpVersion;
 
-                bool caseMatches;
+                if (requestUrlPath.Length > 0 && requestUrlPath[0] != '/')
+                {
+                    int hostIndex;
+                    if (requestUrlPath.StartsWith("http://"))
+                    {
+                        hostIndex = 7;
+                    }
+                    else if (requestUrlPath.StartsWith("https://"))
+                    {
+                        hostIndex = 8;
+                    }
+                    else
+                    {
+                        ReportCorruptedHttpRequest(new BadHttpRequestException($"Invalid request path: {requestUrlPath}"));
```

Comment:
Also do you want play back user input security wise?

---
In reply to: [57367957](https://github.com/aspnet/KestrelHttpServer/pull/711#discussion_r57367957) [](ancestors = 57367957)